lib/app/injection/injection.dart

```diff
@@ -53,6 +55,21 @@ Future<void> initAPIService() async {
   );
 }
 
+/// Check if there is a user extension set in the shared preferences.
+/// If there is, return the country of the user extension.
+/// If there is not, return the country of the device.
+Future<String> _checkCountry() async {
+  final prefs = await SharedPreferences.getInstance();
+
+  if (prefs.containsKey(UserExt.tag)) {
+    final userExtString = prefs.getString(UserExt.tag);
+    final user =
+        UserExt.fromJson(jsonDecode(userExtString!) as Map<String, dynamic>);
+    return user.country;
```

Comment:
Does this work for a temp user? And do we maybe want to check if country really is filled?

Reply:
It works for a temp user, I tested it before